### PR TITLE
Add Outreachy winter 2023 round

### LINKED ISSUE #2158: Update Outreachy information w/ previous round
The [OCaml Outreachy](https://ocaml.org/outreachy) page keeps track of all of the Outreachy internships that the community has taken part in. We should add the "December 2023" round to this list by updating: https://github.com/ocaml/ocaml.org/blob/main/data/outreachy.yml 

The presentation has already been uploaded to watch.ocaml.org and can be linked with timestamps for each of the projects: https://watch.ocaml.org/w/b7sv1LQSVZQH6trf4xpwFX

### PR DESCRIPTION
**Pull request by patricoferris**
Fixes #2158 -- would be good to double-check the information is correct

### CODE REVIEW

File: data/outreachy.yml [hunk 1]
```diff
@@ -299,3 +299,91 @@ rounds:
...
+        blog: "https://dev.to/burnleydev"
+        mentors:
+          - Sayo Bamigbade
+          - punchagan
+        video: https://watch.ocaml.org/w/b7sv1LQSVZQH6trf4xpwFX?start=21m12s
+      - title: "Improve the GUI Experience for OCaml Users"
+        description: >
+          Inspired by [Rust's "Are we GUI yet?"](https://areweguiyet.com/), we want the same work done on the
+          OCaml GUI libraries. A similar work has been done for the OCaml web
+          libraries: ["Is OCaml web yet?"](https://ocaml.org/docs/is-ocaml-web-yet) (see the pull request). This work would
```
Comment (christinerose) [inline data/outreachy.yml:370]: I can only find a pull request for "Is OCaml GUI yet?"

Comment (patricoferris) [reply data/outreachy.yml:370]: Ah, I was copying the descriptions from the original Outreachy pages so I think they predate "Is OCaml web yet?" being merged! https://ocaml.org/docs/is-ocaml-web-yet :))

Comment (patricoferris) [reply data/outreachy.yml:370]: Thanks for the editing @christinerose -- In 2fc2173 I made the descriptions support markdown so these links you added would be rendered on the web page :))

**Review by christinerose — APPROVED** [review]
Looks great!

File: data/outreachy.yml [hunk 2]
```diff
@@ -299,3 +299,91 @@ rounds:
...
+          functional programming concepts with very little reference to Python
+          syntax.
+
+
+          This project aims to implement a geometric creative coding library in
+          OCaml. It is heavily inspired by Joy. When done, it will serve as a
+          means to do geometric creative coding in OCaml.
+        mentee: "Fay Carsons"
+        source: https://github.com/Sudha247/ocaml-joy
+        blog: "https://www.faycarsons.xyz"
```
Comment (cuihtlauac) [inline data/outreachy.yml]:
```suggestion
        blog: "https://www.faycarsons.xyz/articles#joy"
```

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

**Review by cuihtlauac — APPROVED** [review]
This looks good to me

**Pull request merged**